test/resources/examples/index.js

```diff
@@ -203,6 +203,34 @@ const stableXExample = generateTestCase({
   ],
 })
 
+const marginalTrade = generateTestCase({
+  name: "Marginal Trade",
+  orders: [
+    { sellToken: 0, buyToken: 1, sellAmount: feeAdded(toETH(20)).add(ERROR_EPSILON), buyAmount: toETH(10), user: 0 },
+    { sellToken: 1, buyToken: 0, sellAmount: toETH(10), buyAmount: feeSubtracted(toETH(20)).sub(ERROR_EPSILON), user: 1 },
+    { sellToken: 0, buyToken: 1, sellAmount: feeAdded(toETH(200000)).add(ERROR_EPSILON), buyAmount: toETH(100000), user: 2 },
+    {
+      sellToken: 1,
+      buyToken: 0,
+      sellAmount: toETH(100000),
+      buyAmount: feeSubtracted(toETH(200000)).sub(ERROR_EPSILON),
+      user: 3,
+    },
+  ],
+  solutions: [
+    {
+      name: "First Solution",
+      prices: [1, 2].map(toETH),
+      buyVolumes: [toETH(10), feeSubtracted(toETH(20)), ZERO, ZERO],
+    },
+    {
+      name: "Marginally Solution",
```

Comment:
Is the order data here the same as for another (e.g. basic trade)? 
I wonder if it would be better to have a single example which might have a bunch of solutions (basic, better, marginally better, etc.).

Reply:
It’s not quite the same since two of the orders are 10k times bigger, but I could try and fiddle with what we have for examples and make this suggestion happen!

Reply:
Furthermore, if we squeeze all the different orders necessary into one example, then our solution's buyAmount's vector would get large (since `generateTrade` asserts that number of orders is equal to size of solution vector). I am not completely opposed to this, just pointing out some hurdles we will face along the way. I will create an issue for this point and see how it goes!